src/curves/g2curve_target.rs

```diff
@@ -106,6 +109,34 @@ impl<F: RichField + Extendable<D>, const D: usize> G2Target<F, D> {
 
         Self { x, y }
     }
+
+    pub fn pow_var_simple(&self, builder: &mut CircuitBuilder<F, D>, s: &FrTarget<F, D>) -> Self {
+        let bits = builder.split_nonnative_to_bits(&s.target);
+
+        let mut doubles = vec![];
+        let mut v = self.clone();
+        doubles.push(v.clone());
+        for _ in 1..bits.len() {
+            v = v.double(builder);
+            doubles.push(v.clone());
+        }
+
+        assert_eq!(bits.len(), doubles.len());
+
+        let mut rng = rand::rngs::StdRng::seed_from_u64(42);
+        let rando = G2Affine::rand(&mut rng);
+        let rando_t = G2Target::constant(builder, rando);
+        let neg_rando = G2Target::constant(builder, -rando);
+        let mut r = rando_t;
```

Comment:
Is this constant 42 chosen on purpose ? Could you say why it was chosen and put it as a constant ?
Also the -rando is not proven to be the negation of rando. Isn't it necessary to prove they're both inversion of each other ?

Reply:
Oops it's already merged... Actually I just copied this from [G1Target](https://github.com/Lagrange-Labs/plonky2-bn254/blob/ee1e3b6d21a32069034f1bc984bfc3ed63832e22/src/curves/g1curve_target.rs#L141-L168). I'm not sure but I'll update them if I found any issues from this.